# masters-course-2024-genome-bioinformatics

### Lecture 1: Introduction to Genome Bioinformatics
- Overview of Genome Bioinformatics
- An introduction to the scientific method
- Key challenges and applications
- Introduction to genomic data types and sources

### Lecture 2: Genomic Databases and Data Retrieval
- Navigating genomic databases (NCBI, Ensembl, UCSC)
- Methods for searching and downloading genomic data
- Understanding and handling different data formats

### Lecture 3: Basic Genomic Data Processing
- Introduction to command-line tools and scripting
- Data quality assessment and preprocessing
- Introduction to Linux and Bash scripting for bioinformatics

### HOMEWORK 1. [QUAST and BUSCO](https://github.com/aglabx/masters-course-2024-genome-bioinformatics/blob/main/homeworks/hw1_2024.md) (deadline March 6, 2024)

### Lecture 4: Genome Assembly and Assessment
- Principles of genome assembly
- Tools for genome assembly
- Assessing assembly quality and completeness (e.g., QUAST, BUSCO)

Homework 1. Assembly and quality control.

### Lecture 5: Computational Resources and High-Performance Computing
- Introduction to high-performance computing (HPC) and cloud resources
- Using Slurm for job management
- Running bioinformatics workflows on cloud platforms (AWS, Google Cloud)

### Lecture 6: Repetitive DNA and Genome Masking
- Understanding repetitive DNA elements
- Tools for repeat identification and masking (e.g., RepeatMasker, RepeatModeler)
- Implications of repetitive DNA in genomic analysis
- WindowMasker, TRF, dust, and other tools

Homework 2. Repeat masking and annotation.

### Lecture 7: De Novo Gene Prediction and Annotation
- Principles of gene prediction and annotation
- Tools and methods for de novo gene prediction (e.g., AUGUSTUS, GeneMark)
- Training gene prediction models with known data

Homework 3. Structural annotation 1. Gene prediction.

### Lecture 8: Transcriptomics and Iso-Seq Technology
- Overview of transcriptomics and Iso-Seq methodology
- Advantages and challenges of long-read sequencing
- Applications of Iso-Seq in gene annotation and discovery

### Lecture 9: Iso-Seq Data Processing and Quality Control
- Preprocessing and quality assessment of Iso-Seq data
- Tools and workflows for Iso-Seq data (e.g., IsoSeq3, Lima)

### Lecture 10: Advanced Gene Annotation Techniques
- Incorporating transcriptomic and proteomic evidence in annotation
- Comparative genomics and phylogenetics

Homework 4. Structural annotation 2. Evidence based genes prediction.

### Lecture 11: Protein Prediction and Functional Analysis
- Predicting proteins from transcript data
- Functional annotation of proteins (e.g., InterProScan, BLAST)
- Introduction to protein-protein interaction networks

Homework 5. Functional annotation.

### Lecture 12: Comparative Genomics and Evolutionary Analysis
- Exploring the principles and methods used in comparative genomics to understand evolutionary relationships and functional genomics.
- Tools and databases for comparative genomics analysis.
- Case studies on how comparative genomics has elucidated gene function, gene families, and evolutionary histories.

### Lecture 13: Variant Calling Techniques and Tools
- Introduction to Genetic Variants: Brief overview of different types of genetic variants (SNPs, indels, CNVs, SVs) and their biological significance.
- Principles of Variant Calling: Understanding the workflow from raw sequencing data to variant identification, including alignment, base quality score recalibration, and variant calling.
- Key Tools and Algorithms: Overview of popular variant calling tools (e.g., GATK, Samtools, FreeBayes) and their comparative strengths and use cases.
- Quality Assessment: Strategies for assessing the quality of called variants, including variant quality score recalibration, filtering criteria, and validation techniques.
- Challenges and Solutions: Discussing common challenges in variant calling, such as calling variants in low-complexity regions, dealing with highly polymorphic regions, and differentiating between true variants and sequencing artifacts.

### Lecture 14: Applications of Called Variants
- Genomic Association Studies: How called variants are used in GWAS to identify genetic variants associated with diseases or traits.
- Pharmacogenomics and Personalized Medicine: Understanding the role of variants in drug response variability and their implications in developing personalized treatment plans.
- Functional Genomics: Methods to predict the functional impact of genetic variants on gene expression, protein function, and phenotype using bioinformatics tools (e.g., SIFT, PolyPhen-2).
- Population Genetics and Evolution: Analyzing genetic variation across populations to study human evolution, migration patterns, and population-specific disease predispositions.
- Clinical Diagnostics and Genetic Screening: Use of variant calling in diagnosing genetic disorders, carrier screening, and prenatal testing. Discussion on the ethical, legal, and social implications of genetic testing.

Homework 6. Variants annotation.

### Lecture 16: Data Visualization and Interpretation
- Tools and strategies for data visualization (e.g., JBrowse, UCSC Genome Browser)
- Interpreting and presenting bioinformatics analysis results
- Ethical considerations and data sharing

Homework 7. TBA.

### Lecture 17: TBA

### Lecture 18: TBA

Homework 8. TBA.

### Lecture 19: TBA

### Lecture 20: TBA

### Lecture 21: TBA

### Lecture 22: TBA

### Lecture 23: TBA
