### Genome Bioinformatics: Homework 1. Genome Downloading and Quality Control

**Course Code:** Special topics in genomics

**Instructor:** Aleksey Komissarov

**Submission Date:** March 6, 2024

**File location:** https://github.com/aglabx/masters-course-2024-genome-bioinformatics/blob/main/homeworks/hw1_2024.md

#### Objective:
This assignment is designed to offer you a comprehensive, hands-on experience in genome bioinformatics, aimed at enhancing your understanding and analytical skills in several key areas. Throughout a series of tasks, you will:

- Search for and download genomes from the NCBI database, learning how to navigate and extract relevant genomic data.
- Perform a sanity check for adapter contamination in the data used.
- Conduct a sanity check for prokaryotic DNA contamination.
- Assess the quality of genome assembly using QUAST, gaining insights into its quantitative characteristics.
- Evaluate the quality of genome assembly using BUSCO, understanding the structural integrity and completeness of gene content.

Completing these tasks will equip you with a solid set of skills for manipulating, analyzing, and interpreting genomic data, preparing you for advanced research and challenges in genome bioinformatics.

#### Instructions:

Complete the following tasks and submit your work to your private GitHub repository according to the provided instructions for each step below. Ensure all data and analyses are clearly labeled and organized as described in the instructions.

---

### Task 0. Preparation for Homework Assignments (1 point)

#### Instructions:

1. **Create a Private Repository**: Name the repository “masters-course-2024-genome-bioinformatics”. This centralized repository will be used to organize and submit all your homework assignments for this course.
   
2. **Add Collaborator**: Navigate to the repository settings, find the "Collaborators and Teams" section, and add me (@ad3002) as a collaborator. This step is crucial for enabling your instructor to review and grade your submissions.

3. **Organize Your Work**: For each homework assignment, create a dedicated folder within the repository. Name these folders according to the assignment number (e.g., hw1, hw2, etc.). This organization will help keep your work orderly and easily accessible.

4. **Report file location**: Create a folder called `hw1` in your repository. This is where you will submit your homework assignment for this task. The README.md file in this folder will contain the report for this task.

#### Deliverables:

- **Private Repository Creation**: Successfully create a private repository named “masters-course-2024-genome-bioinformatics”.

- **Collaborator Addition**: Add @ad3002 to the repository as a collaborator, ensuring that the instructor has access to review your work.

#### Suggestions for Improvement:

- **Include a README File**: In the root directory of your repository, consider adding a README.md file. This file should briefly describe the repository's purpose, its structure, and any relevant instructions for navigating through your homework submissions. A well-documented README file can significantly enhance the clarity and professionalism of your repository.

- **Use Descriptive Commit Messages**: When submitting or updating your homework assignments, use clear and descriptive commit messages. This practice helps track changes effectively and understand the nature of each update.

- **Open Communication via Issues**: Encourage the use of the repository's issue tracker for questions about homework, requests for clarifications, or suggestions for corrections and improvements. This proactive approach promotes a collaborative and supportive learning environment, allowing for the exchange of ideas and resolutions to any challenges encountered (it will give you additional points).

---

### Task 1: Searching and Downloading Genomes  (2 points)

**Objective:** Familiarize yourself with genomic databases and acquire genomic data.

#### Instructions:

1. **Selecting Organisms:**
   - Choose one Brassicaceae species or any other organism.
   - It is recommended to select an organism with the refseq annotation to facilitate comprehensive analysis. Otherwise, you will significantly increase the hardness of the task.
   - It is recommended to select an organism with a relatively small genome size to facilitate the analysis. Otherwise, you will significantly increase the hardness of the task.
   - You have two stragegies: 
     - Take more engeeniering approach and select a model organism with a well-annotated genome. This will allow you to use the genome for a wide range of bioinformatics tasks. And your primary goal will be to understand the genome structure and the annotation process. The final deliverable will be a comprehensive genome annotation.
     - Take more biological approach and select a non-model organism with a less annotated genome but some interesting biological features. It will make it more difficult to work with the genome, but it will allow you to learn more about the genome assembly and annotation process. The final deliverable will be not only a comprehensive genome annotation but a set of biological insights about the organism formulated as one or several hypotheses that can be tested in the future with wet-lab experiments.
   - Provide a brief description of the organism and why you chose it in the report (hw1/README.md).

2. **Database Navigation:**
   - **Objective:** Locate the genomic data for your selected organisms across multiple databases.
   - **Databases to Explore:**
     - **NCBI (National Center for Biotechnology Information):** A comprehensive resource offering a wide variety of genomic data.
     - **Ensembl:** A database that provides genomic annotations and visualization tools.
     - **UCSC Genome Browser:** An interactive web-based tool offering a detailed view of genomic data.

   - **Instructions:**
     - **Locating Genomes:**
       - For each database (NCBI, Ensembl, UCSC), search for the genome of your selected  organism.
       - Record the search terms used and any filters applied during your search.
       - Add to the report a full path to the found genome file in the database and the file name. Or note the your genome is absent in one of the databases.

     - **Comparison:**
       - Once located, compare the genomes from each database by considering the following aspects:
         - **File Size:** Document the file size for each genome. Are they significantly different across databases?
         - **Number of Lines:** Use a tool to count the number of lines in each genome file. This can provide insights into the file's structure and content.
         - **Number of Headers:** Count and compare the number of headers (usually denoted by '>' in FASTA files). This can indicate the number of sequences or chromosomes.
         - **Header Format:** Examine and describe the format of the headers. Are they consistent across databases? Do they provide different types of information?
         - Make a table in your report with the comparison of the genomes from different databases (https://www.codecademy.com/resources/docs/markdown/tables).

   - **Analysis:**
     - Reflect on the similarities and differences observed between the genomes from different databases. Consider why discrepancies might exist, such as different assembly versions or naming conventions.
     - Think how these differences might impact genomic research and analysis.

#### Deliverables:
- The report file should include a detailed description of the selected organism and the reasons for choosing it.
- he report file should include a summary of the search terms and filters used to locate the genome in each database.
- The report should contain a comparison table of the genomes from different databases.

---

### Task 2: Running QUAST  (3 points)
**Objective:** Assess the quality of genome assemblies using QUAST to understand their structural integrity and completeness.

#### Instructions:
1. **Installation:**
   - **Objective:** Install QUAST using the simplest method to familiarize yourself with software installation and environment management in bioinformatics.
   - **Conda Installation:**
     - Use the Conda package manager to install QUAST on your local or server machine. Conda can simplify software installation and manage dependencies effectively.
     - Provide a detailed summary of the installation process, including any conda commands used.

2. **Running QUAST:**
   - **Objective:** Execute QUAST to analyze the quality of the downloaded genomes and understand the implications of various parameters and options.
   - **Standard Analysis:**
     - Run QUAST on the downloaded genomes. Document the command used, including any parameters or options.
   - **Considerations for Large Eukaryotic Genomes:**
     - Google about the specific QUAST option recommended for large eukaryotic genomes. Explain why this option is necessary and how it alters the analysis.
     - Mention any options or parameters that should typically not be used for large genomes due to computational or accuracy concerns.

3. **Interpreting Results:**
   - **Objective:** Develop the ability to understand and interpret the output from QUAST to assess genome assembly quality.
   - **Summary of Key Metrics:**
     - Provide a summary of the output, focusing on key metrics like N50, L50, and total length. Describe what each metric means.
   - **Analysis:**
     - Explain what these metrics indicate about the quality of each genome assembly. Discuss any potential concerns or noteworthy points revealed by the metrics.
     - Reflect on how these quality assessments might impact subsequent genomic analysis and research.

#### Deliverables:
- The report should containt the installation process commands, the commands used to run QUAST (see about code blocks https://www.codecademy.com/resources/docs/markdown/code-blocks).
- The report should containt an analysis of what the QUAST metrics reveal about the genomic data you're working with.
- The quast report file in txt format as **hw1/quast_report.txt**.

**Note:** Ensure your report is well-organized, with each section clearly labeled and explained. Your reflections and interpretations should demonstrate a deep understanding of the role and importance of genome assembly quality assessment in bioinformatics.

---

### Task 3: Running BUSCO  (4 points)

**Objective:** Assess the quality and completeness of genome assemblies or annotations using BUSCO to understand the presence and completeness of universal single-copy orthologs.

#### Instructions:

**Installation:**

**Objective:** Install BUSCO using conda to familiarize yourself with software installation and environment management in bioinformatics. Don't forget to create and activate the separate environment before installing BUSCO.

- **Conda Installation:**
  - Use the Conda package manager to install BUSCO on your local or server machine. Conda simplifies software installation and manages dependencies effectively.
  - Provide a detailed summary of the installation process, including any conda commands used.

**Running BUSCO:**

**Objective:** Execute BUSCO to analyze the quality of downloaded genomes or annotations and understand the implications of various parameters and options.

- **Standard Analysis:**
  - Run BUSCO on the downloaded genomes or annotations. Document the command used, including any parameters or options.

- **Choosing the Appropriate Lineage:**
  - Choose the appropriate lineage dataset for your organism. Explain how to select the appropriate lineage dataset and how this choice impacts the analysis (start here https://busco.ezlab.org/frames/euka.htm).

- **Lineage-Specific Analysis:**
  - Discuss the importance of selecting the appropriate lineage dataset for your organism. Explain how this choice impacts the analysis.

- **Advanced Options:**
  - Mention any advanced options or parameters that are particularly useful for your dataset. This might include parallel running options for large datasets or specific parameters for different types of analysis (genomes, proteins, transcriptome).

**Interpreting Results:**

**Objective:** Develop the ability to understand and interpret the output from BUSCO to assess genome assembly or annotation completeness.

- **Summary of Key Metrics:**
  - Provide a summary of the output, focusing on key metrics like Complete BUSCOs, Fragmented BUSCOs, and Missing BUSCOs. Describe what each metric means.

- **Analysis:**
  - Explain what these metrics indicate about the quality and completeness of each genome assembly or annotation. Discuss any potential concerns or noteworthy points revealed by the metrics.

- **Contextualizing Results:**
  - Reflect on how these completeness assessments might impact subsequent genomic analysis and research. Discuss the role of evolutionary expectations in assessing gene content.

#### Deliverables:

- **A comprehensive report detailing the installation process, the commands used to run BUSCO, and a thorough interpretation of the results.**
  - See notes about quast, we expected that you will provide an installation process commands, the commands used to run BUSCO, and an analysis of what the BUSCO metrics reveal about the genomic data you're working with as code blocks in your report.
  - The busco report file in txt format as **hw1/short_summary.specific.XXX_odb10.busco_dmix_rLacAgi1.txt**, where XXX is your library name.

**Note:** Ensure your report is well-organized, with each section clearly labeled and explained. Your reflections and interpretations should demonstrate a deep understanding of the role and importance of genome assembly or annotation completeness assessment in bioinformatics.

---

### Expected files in hw1 folder:
- README.md
- quast_report.txt
- short_summary.specific.XXX_odb10.busco_dmix_rLacAgi1.txt

### About the report file (README.md):
- Motivation for selecting the organism.
- Summary of the search terms and filters used to locate the genome in each database.
- Comparison table of the genomes from different databases.
- Installation process commands for QUAST and BUSCO.
- The commands used to run QUAST and BUSCO.
- An analysis of what the QUAST metrics reveal about the genomic data you're working with.
- An analysis of what the BUSCO metrics reveal about the genomic data you're working with.


### Pitfalls to Avoid:
- Always use separate environments for software installation and testing. Use conda create / activate to create and activate a new environment for each complex software installation. Deactivate the environment when you are done. Don't install software in your base environment except for very simple tools.
- Contral remaining disk space on your server. Use the `df -h` command to check the remaining disk space on your server. If you are running out of disk space, delete unnecessary files or move them to a different location.
- Don't use the home directory for storing your data.
- Back up your data/code regularly. Use the `rsync` command to back up your data to a different location.
- Don't use the root directory for chaotically storing your data. Use folders and subfolders to organize your data. For example, create a folder called `data` and then create subfolders for each task. Or create a folder for each task and then create subfolders for each dataset. And be consistent with your naming/structure conventions.
- Good practice to check the size of your data before downloading it. Use the `du -sh` command to check the size of your data.
- You can check file contents using the `head` and `tail` commands. For example, `head -n 10 file.txt` will print the first 10 lines of the file. `tail -n 10 file.txt` will print the last 10 lines of the file. Sometimes it will save a lot of time with some strange errors when it was wrong file or wrong format.
- Remember about --threads and --pa flags, don't run heavy tasks with only one thread.
- Please use Screen / tmux and similar for running processes in the background, don't use nohup.
- Please download genome as one file, don't use individual 

### About LabJournal:
- LabJournal is a tool for organizing your research. It is a simple, yet powerful tool for keeping track of your research. It is a great tool.

**Good luck, and enjoy your journey through the world of genome bioinformatics!**

